pages/product/[name].js

```diff
@@ -0,0 +1,117 @@
+import Link from 'next/link'
+import NextImage from 'next/image'
+import { useState, useEffect } from 'react'
+import { StarIcon } from '@heroicons/react/solid'
+import { relativizeURL, shimmer, toBase64 } from '@/components/Shimmer'
+import { HeartIcon, StarIcon as StarIconOutline } from '@heroicons/react/outline'
+
+const Product = ({ data }) => {
+  const [relatedProducts, setRelatedProducts] = useState([])
+  const [selectedImage, setSelectedImage] = useState(data.images[0].url)
+
+  // Fetch the items for listing at the bottom
+  useEffect(() => {
+    fetch('/l0-api/products/all')
+      .then((res) => res.json())
+      .then((res) => {
+        if (res && res.length > 0) {
+          setRelatedProducts(res)
+        }
+      })
+  }, [])
+
+  useEffect(() => {
+    setSelectedImage(data.images[0].url)
```

Comment:
Rather than setting the whole image element into state, can't we just set the active index? Also with your use effect here on data, any update to the tree will immediately change the image that's in view. This `useEffect` seems unnecessary.

Reply:
- We can, let me do that.
- So if you're on a PDP, scroll to the bottom and click on any other product, the main image doesn't change because the first time component mounted, the image was set. In case any changes are made to the server side data (that is someone clicking for another PDP while being on a PDP), using the useEffect I change the image.

Reply:
Done, handled with index.